File: 441-arranging-coins/arranging-coins.java
class Solution {
    public int arrangeCoins(int n) {
        int l = 0;
        int h = n;
        while(l<=h)
        {
            int mid = l + (h-l)/2;
            long sum =(long)mid*(mid+1)/2;
            if(sum==n) return mid;
            else if(sum<n)l = mid+1;
            else h = mid-1;
        }
        return h;
    }
}

File: 441-arranging-coins/arranging-coins.cpp
class Solution {
public:
    int arrangeCoins(int n) {
        int low = 0;
        int high = n;
        int ans = 0;
        while(low<=high)
        {
            int mid = (low+(high-low)/2);
            long val = (long)mid*(mid+1)/2;
            if(n==val) return mid;
            else if(val < n){
                ans = max(mid,ans);
                low = mid+1;
            }
            else  high = mid-1;
        }

        return ans;
    }
};